tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -1413,6 +1413,72 @@ def create_full_val_real_complex128(val):
     assert(isclose(     f_real_complex128(val_float)       ,      create_full_val_real_complex128(val_float), rtol=RTOL, atol=ATOL))
     assert matching_types(f_real_complex128(val_float), create_full_val_real_complex128(val_float))
 
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = [
+            pytest.mark.skip,
+            pytest.mark.python]
+        ),
+    )
+)
+
+def test_full_dtype_auto(language):
+    @types('int')
+    @types('float')
+    @types('complex')
+    @types('int32')
+    @types('float32')
+    @types('float64')
+    @types('complex64')
+    @types('complex128')
+    def create_full_val_auto(val):
+        from numpy import full
+        a = full(3,val)
+        return a[0]
+
+    integer   = randint(low = iinfo('int').min,   high = iinfo('int').max,   dtype=int)
+    integer32 = randint(low = iinfo('int32').min, high = iinfo('int32').max, dtype=np.int32)
```

Comment:
Values like `iinfo('int').min` are already defined at the top of this file. See [here](https://github.com/pyccel/pyccel/blob/master/tests/epyccel/recognised_functions/test_numpy_funcs.py#L11-L33). Please use them rather than recalculating them